### PR TITLE
[daint,dom] Add spack-config module

### PR DESCRIPTION
**Pull request by haampie**

### CODE REVIEW

**Comment by haampie** [comment]
@lucamar this module should make https://github.com/eth-cscs/production/tree/master/spack mostly redundant, and a bit more convenient, so you can do `module load spack-config` s.t. the user's spack instance becomes aware of system packages (through en variable + generated config files).

Question I have: what's the naming convention for these files? There's not toolchain required, just git / python. The cdt number is hard-coded. It doesn't work on Eiger, cause that needs `cpe-...` not `cdt-...` apparently.

Also, is this module automatically updated when the cdt version is bumped?

**Review by ekouts — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/spack-config/spack-config-1.0.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+modtclfooter = """
+set SPACK_CDT_CONFIG_ROOT "%(installdir)s/%(_cdt)s"
+
+if { [file exists $SPACK_CDT_CONFIG_ROOT] == 1 } {
+  setenv SPACK_SYSTEM_CONFIG_PATH "$SPACK_CDT_CONFIG_ROOT"
+} else {
+  puts stderr "$SPACK_CDT_CONFIG_ROOT does not exist"
+  exit 1
+}
+"""
```
[the remark this answers is not in the document]
Comment (ekouts) [reply easybuild/easyconfigs/s/spack-config/spack-config-1.0.eb:39]: The double "%%" is required in order to mix template values with your own variables.

[the remark this answers is not in the document]
Comment (haampie) [reply easybuild/easyconfigs/s/spack-config/spack-config-1.0.eb]: Thanks!

**Review by ekouts — APPROVED** [review]

**Comment by haampie** [comment]
@ekouts am I supposed to open separate prs cause dom does not have 20.10 and daint does not have 21.09?

**Comment by haampie** [comment]
Or can you show me how to just stick to 1  `eb` file for all the systems? The only variable thing is `cdt-x.y` on dom/daint and `cpe-x.y` on eiger/pilatus.

**Comment by haampie** [comment]
Hm, I've just created two separate prs then #2548  and #2549

**Comment by haampie** [comment]
Closing in favor of the other two

**Pull request closed without merging**